nistats/thresholding.py

```diff
@@ -10,48 +10,98 @@
 
 
 def fdr_threshold(z_vals, alpha):
-    """ return the BH fdr for the input z_vals"""
+    """ return the Benjamini-Hochberg FDR threshold for the input z_vals
+    
+    Parameters
+    ----------
+    z_vals: array,
+            a set of z-variates from which the FDR is computed
+    alpha: float,
+           desired FDR control
+    
+    Returns
+    -------
+    threshold: float,
+               FDR-controling threshold from the Benjamini-Hochberg procedure
+    """
+    if alpha < 0 or alpha > 1:
+        raise ValueError('alpha should be between 0 and 1')
     z_vals_ = - np.sort(- z_vals)
     p_vals = norm.sf(z_vals_)
     n_samples = len(p_vals)
     pos = p_vals < alpha * np.linspace(
         .5 / n_samples, 1 - .5 / n_samples, n_samples)
     if pos.any():
-        return (z_vals_[pos][-1] - 1.e-8)
+        return (z_vals_[pos][-1] - 1.e-12)
     else:
         return np.infty
 
 
-def map_threshold(stat_img, mask_img=None, threshold=.001,
+def map_threshold(stat_img=None, mask_img=None, level=.001,
                   height_control='fpr', cluster_threshold=0):
-    """ Threshold the provided map
+    """ Compute the required threshold level and return the thresholded map
 
     Parameters
     ----------
-    stat_img : Niimg-like object,
+    stat_img : Niimg-like object or None, optional
        statistical image (presumably in z scale)
+       whenever height_control is 'fpr' or 'none', 
+       stat_img=None os acceptable. 
```

Comment:
is

Reply:
May be: "For height_control='fpr', stat_img has no effect and can be given as None."

Reply:
It is actually a bit more subtle: when the image is provided, a thresholded version is returned. For this readons, I prefer to keep the current formulation.